src/client/index.ts

```diff
@@ -18,111 +79,268 @@ export interface NotificationOptions {
+     * - `"Event Modified"`: When an event is modified
+     * - `"Event Cancelled"`: When an event is cancelled
+     * - `"Event Response"`: An attendee has responded to an event invitation that you created
+     * - `"Daily Agenda"`: A notification sent each morning with event reminders
+     *
+     * **NOTE:** The user-facing UI that a user would use to manage their preferences is still in progress. This
+     * property has been added in advance of this UI being released, to ensure future compatibility.
      */
-    subtitle?: string;
+    category: string;
+
     /**
-     * URL of the icon to be displayed in the Notification.
+     * URL of the icon to be displayed in the notification.
      */
     icon?: string;
+
     /**
-     * Any custom context data associated with the Notification.
+     * Any custom context data associated with the notification.
      */
     customData?: CustomData;
+
     /**
-     * The timestamp shown on the Notification.  This is presentational only - a future date will not incur a scheduling action.
+     * The timestamp shown on the notification. If omitted, the current date/time will be used.
+     *
+     * This is presentational only - a future date will not incur a scheduling action.
      */
     date?: Date;
+
     /**
-     * Text and icons for up to two Notification action buttons.
+     * A timestamp at which point the notification will be removed from the user's notification center.
+     *
+     * If omitted or `null`, notification will persist until user explicitly actions/clears it.
      */
-    buttons?: OptionButton[];
-}
+    expires?: Date|null;
 
-/**
- * Configuration options for constructing a Button within a Notification.
- */
-export interface OptionButton {
-    title: string;
-    iconUrl?: string;
+    /**
+     * A list of button definitions, to display below the notification text.
+     *
+     * Notifications support up to four buttons, attempting to add more than four will result in the notification
+     * being rejected by the service.
+     */
+    buttons?: ButtonOptions[];
+
+    /**
+     * An action that will fire when the notification is clicked. This only fires on clicks to the notification body,
+     * interactions with buttons (both application-defined buttons, and the default 'X' close button) will not trigger
+     * a select action.
+     *
+     * If omitted or `null`, applications will not receive a {@link NotificationActionEvent|`notification-action`}
+     * event when the notification is clicked. See {@link Actions} for more details on Notification actions, and
+     * receiving interaction events from notifications.
+     */
+    onSelect?: ActionDeclaration<never, never>|null;
 }
 
 /**
- * User-defined context data that can be attached to Notifications.
+ * Application-defined context data that can be attached to notifications.
  */
 export type CustomData = any;
```

Comment:
Should we enforce this is an object and not a function?

Reply:
Depends... Might be a PITA for people using it.

Declaring it as `{}` or `{}|boolean|number|string` means that you can no longer use `customData.foo` syntax, you have to use `customData['foo']` to avoid compile errors.

Reply:
Was forgetting the significance of the "and not a function" part.

Agree, have updated.